cpp/src/arrow/array/builder_adaptive.h

```diff
@@ -122,9 +122,10 @@ class ARROW_EXPORT AdaptiveUIntBuilder : public internal::AdaptiveIntBuilderBase
 
   std::shared_ptr<DataType> type() const override;
 
+  Status ExpandIntSize(uint8_t new_int_size);
```

Comment:
Hmm... I don't think it is required to make this public. Is it?

Reply:
This function needn't be public anymore.